frontend/src/components/mine/Incidents/MineIncidents.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from "react";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import CustomPropTypes from "@/customPropTypes";
+import { createDropDownList } from "@/utils/helpers";
+import * as Permission from "@/constants/permissions";
+import * as ModalContent from "@/constants/modalContent";
+import { modalConfig } from "@/components/modalContent/config";
+import AddButton from "@/components/common/AddButton";
+import AuthorizationWrapper from "@/components/common/wrappers/AuthorizationWrapper";
+
+import { fetchMineIncidents, createMineIncident } from "@/actionCreators/mineActionCreator";
+import { getMineIncidents } from "@/selectors/mineSelectors";
+import { getIncidentFollowupActionOptions } from "@/selectors/staticContentSelectors";
+
+import { MineIncidentTable } from "./MineIncidentTable";
+
+/**
+ * @class  MineTailingsInfo - all tenure information related to the mine.
+ */
+
+const propTypes = {
+  mine: CustomPropTypes.mine.isRequired,
+  mineIncidents: PropTypes.arrayOf(CustomPropTypes.incident),
+  followupActions: PropTypes.arrayOf(CustomPropTypes.incidentFollowupType),
+  openModal: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  fetchMineIncidents: PropTypes.func.isRequired,
+  createMineIncident: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  mineIncidents: [],
+  followupActions: [],
+};
+
+export class MineIncidents extends Component {
+  componentDidMount() {
+    this.props.fetchMineIncidents(this.props.mine.mine_guid);
+  }
+
+  handleAddMineIncident = (values) => {
+    this.props.createMineIncident(this.props.mine.mine_guid, values).then(() => {
+      this.props.closeModal();
+      this.props.fetchMineIncidents(this.props.mine.mine_guid);
+    });
+  };
+
+  openMineIncidentModal = (event) => {
+    event.preventDefault();
+    this.props.openModal({
+      props: {
+        onSubmit: this.handleAddMineIncident,
+        title: ModalContent.ADD_INCIDENT(this.props.mine.mine_name),
+        mineGuid: this.props.mine.mine_guid,
+        followupActionOptions: createDropDownList(
```

Comment:
It looks like you're already doing this in `frontend/src/selectors/staticContentSelectors.js`. If it's in the Redux Store, you should be able to just access it from there instead of regenerating the dropdown list every time (this is a fairly slow operation for JS)

Reply:
I need to access the object list, and the dropdown converted list, so I stopped using the selector one, cause the form doesn't access the store. 

I'll delete it from staticContentSelectors.js

Reply:
Transforming `followupActionOptions` into a dropdown would be best done in the selector. You can utilize `createSelector` for this, there are a lot of examples in staticContentSelectors.js that do the exact same thing, 


```
export const getDropdownfollowupActionOptions = createSelector(
  [followupActionOptions],
  (options) => createDropDownList(options, "description", "mine_incident_followup_type_code")
);
```
you can import `getDropdownfollowupActionOptions` like you do `followupActionOptions` and pass it into the form the same way.

Reply:
I mean I could do it through a selector, but I already have the list of things I want. Does it make sense to have two props, one for the transformed list and one for the normal list, just so i can hand one of them to a child? just making sure i get it.

Reply:
Yeah, redux state transformations should occur in the selectors.